environs/config/config.go

```diff
@@ -688,6 +718,11 @@ func Validate(cfg, old *Config) error {
 		}
 	}
 
+	// The user shouldn't specify both old and new proxy values.
+	if cfg.HasLegacyProxy() && cfg.HasJujuProxy() {
```

Comment:
I think we should also validate that juju no proxy, if set, contains valid cidr values